core-common/src/main/java/org/zalando/nakadi/domain/EventTypeSchema.java

```diff
@@ -1,9 +1,20 @@
 package org.zalando.nakadi.domain;
 
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import static com.fasterxml.jackson.annotation.JsonTypeInfo.As;
+import static com.fasterxml.jackson.annotation.JsonTypeInfo.Id;
+
 import org.joda.time.DateTime;
 
 public class EventTypeSchema extends EventTypeSchemaBase {
+    @JsonTypeInfo(use = Id.NAME, property = "type", include = As.EXTERNAL_PROPERTY)
+    @JsonSubTypes(value = {
+            @JsonSubTypes.Type(value = JsonVersion.class, name = "JSON_SCHEMA"),
+            @JsonSubTypes.Type(value = AvroVersion.class, name = "AVRO_SCHEMA")
+    })
```

Comment:
@samizzy 
Not that I am against fancy version matching, but it seems that it would be much simpler here to use string representation and create actual object to process logic somewhere in code where it is used. 
There following responsibilities are on this class now: 
1. DTO
2. serialization/deserialization of backend logic through dto
3. backend logic of schema evolution

If we leave only DTO part, than the property type will be string, leaving all the problems with mappings behind

Reply:
I mean make it of type `string`